File: frontend/src/api/market.ts
import api from './index'

export interface Ticker {
  symbol: string
  exchange: string
  last_price: number
  bid: number
  ask: number
  volume_24h: number
  high_24h: number
  low_24h: number
  timestamp: string
}

export interface Kline {
  symbol: string
  exchange: string
  interval: string
  open_time: string
  open: number
  high: number
  low: number
  close: number
  volume: number
  close_time: string
}

export interface OrderBook {
  symbol: string
  exchange: string
  bids: [number, number][]
  asks: [number, number][]
}

export const marketApi = {
  // 获取实时行情
  getTicker: async (symbol: string, exchange = 'binance') => {
    const response = await api.get(`/market/ticker/${symbol}`, {
      params: { exchange }
    })
    return response.data.data as Ticker
  },

  // 获取 K 线数据
  getKlines: async (symbol: string, interval = '1h', limit = 100, exchange = 'binance') => {
    const response = await api.get(`/market/klines/${symbol}`, {
      params: { exchange, interval, limit }
    })
    return response.data.data as Kline[]
  },

  // 获取订单簿
  getOrderBook: async (symbol: string, limit = 20, exchange = 'binance') => {
    const response = await api.get(`/market/orderbook/${symbol}`, {
      params: { exchange, limit }
    })
    return response.data.data as OrderBook
  }
}

File: backend/models/market.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Index
from sqlalchemy.sql import func
from models.user import Base


class Ticker(Base):
    """实时行情快照"""
    __tablename__ = "tickers"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    symbol = Column(String(20), nullable=False, index=True)  # BTCUSDT
    exchange = Column(String(20), nullable=False)  # binance, okx
    last_price = Column(Float, nullable=False)
    bid = Column(Float)
    ask = Column(Float)
    volume_24h = Column(Float)
    high_24h = Column(Float)
    low_24h = Column(Float)
    timestamp = Column(DateTime, default=func.now(), index=True)
    
    __table_args__ = (
        Index('idx_symbol_exchange', 'symbol', 'exchange'),
    )


class Kline(Base):
    """K线数据"""
    __tablename__ = "klines"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    symbol = Column(String(20), nullable=False)
    exchange = Column(String(20), nullable=False)
    interval = Column(String(5), nullable=False)  # 1m, 5m, 1h, 1d
    open_time = Column(DateTime, nullable=False)
    open = Column(Float, nullable=False)
    high = Column(Float, nullable=False)
    low = Column(Float, nullable=False)
    close = Column(Float, nullable=False)
    volume = Column(Float, nullable=False)
    close_time = Column(DateTime, nullable=False)
    
    __table_args__ = (
        Index('idx_kline_lookup', 'symbol', 'exchange', 'interval', 'open_time', unique=True),
    )
